src/components/implementation/no_interface/cFE_booter/osfilesys.h

```diff
@@ -27,41 +36,64 @@ typedef enum {
     FSOBJ_DIR,
 } fsobj_type_t;
 
+struct file_position {
+    struct f_part *open_part;  //part of file being read/written to
+    uint32 part_offset;         //offset into part's data
+    uint32 file_offset;         //position within file as a whole
+    char *addr;
+};
+
 struct fsobj {
     char *name;
-    int32 filedes; // 0 for free file, posative for tracked file
+    int32 filedes; // 0 for free file, positive for tracked file
     fsobj_type_t type;
     size_t size;
-    uint32 refcnt; //number of programs which have opened it
-    char *data;
+    struct file_position position;
+    uint32 refcnt; //number of threads which have opened it
+    uint32 mode;
+    struct f_part *file_part;
     struct fsobj *next, *prev;
-    struct fsobj *child, *parent;   /* child != NULL iff type = dir */
+    struct fsobj *child, *parent; // child != NULL iff type = dir
+};
+
+typedef enum {
+    STATIC,
+    DYNAMIC,
+} fpart_type;
+
+struct f_part {
+    struct fsobj *file;
+    struct f_part *next, *prev;
+    fpart_type memtype;
+    char *data;
 };
 
-struct FS {
+struct fs {
     char *devname;
     char *volname;
     char *mountpoint;
     uint32 blocksize;
     uint32 numblocks;
     struct fsobj *root;
-    struct FS *next, *prev;
 };
 
 /******************************************************************************
 ** Tar Level Methods
 ******************************************************************************/
 
+uint32 FS_entrypoint();
+uint32 FS_posttest();
+uint32 BFT(struct fsobj *o);
```

Comment:
My bad, I needed to do a better job checking before I PR.  BFT was breadth first traversal btw but all it was used for was debugging.